sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -3165,6 +3156,26 @@ private <DestinationT> WriteResult continueExpandTyped(
           storageApiDynamicDestinations =
               new StorageApiDynamicDestinationsBeamRow<>(
                   dynamicDestinations, elementSchema, elementToRowFunction);
+        } else if (getAvroRowWriterFactory() != null) {
```

Comment:
Please add a writeGenericRecords() method so users don't have to deal with the avro format function.

Also - AFAICT from inspection, the AvroWriterFactory is a bit broken if side inputs are used for schemas. Do we have a test for this?

Reply:
> Please add a writeGenericRecords() method so users don't have to deal with the avro format function.
Done.

>AFAICT from inspection, the AvroWriterFactory is a bit broken if side inputs are used for schemas. Do we have a test for this?
You mean when the TableSchema is provided as a PCollectionView? If so I can add a test for that.

Reply:
Yeah - because setSideInputAccessorFromProcessContext needs to be called on the dynamic destinations in every DoFn that might access this, and I'm not sure that's correct for AvroWriterFactory

Reply:
Added a test to check on this for StorageWrites API

Reply:
Also and JIC, besides the added unit test, checked this PR's code with a simple [DynamicDestination definition](https://github.com/prodriguezdefino/apache-beam-streaming-tests/blob/map-support-bq-storagewrites/canonical-streaming-pipelines/src/main/java/com/google/cloud/pso/beam/pipelines/transforms/WriteFormatToBigQuery.java#L70) and seems to work fine for job id: `2023-01-31_12_10_08-12389116961981372457`